doc_src/fish_mode_prompt.txt

```diff
@@ -2,10 +2,36 @@
 
 \subsection fish_mode_prompt-synopsis Synopsis
 
-fish_mode_prompt will output the mode indicator for use in vi-mode.
+The fish_mode_prompt function will output the mode indicator for use in vi-mode.
 
 \subsection fish_mode_prompt-description Description
 
-The output of `fish_mode_prompt` will be displayed in the mode indicator position to the left of the regular prompt.
+The output of `fish_mode_prompt` will be displayed in the mode indicator
+position to the left of the regular prompt. Define your own function to
+customize the appearance of the mode indicator. The `$fish_bind_mode variable`
+can be used to determine the current mode. It will be one of `default`,
+`insert`, `replace_one`, or `visual`.
```

Comment:
This statement is correct. But it worries me because that list of modes can change. For example, the `replace_one` mode was recently added (it had been named `replace-one` until April of this year). Your example also highlights that the `fish_default_mode_prompt` function (which you based your example on) does not deal with an unrecognized mode.

Also, most people who have asked us about customizing the vi mode portion of the prompt want to suppress it completely. So you should include an example showing how defining an empty function will suppress the mode indicator.